kafka_utils/kafka_consumer_manager/util.py

```diff
@@ -194,28 +188,21 @@ def read_groups(self, partition=None):
         if partition is not None:
             self.consumer.assign([TopicPartition(CONSUMER_OFFSET_TOPIC, partition)])
         else:
-            self.consumer.subscribe([CONSUMER_OFFSET_TOPIC])
+            self.consumer.assign(
+                [
+                    TopicPartition(CONSUMER_OFFSET_TOPIC, p)
+                    for p in self.consumer.partitions_for_topic(CONSUMER_OFFSET_TOPIC)
+                ]
+            )
 
         self.log.info("Consumer ready")
         self.watermarks = self.get_current_watermarks(partition)
-        self.retry = 0
         while not self.finished():
-            try:
-                message = self.consumer.next()
-                max_offset = self.get_max_offset(message.partition)
-                if message.offset >= max_offset - 1:
-                    self.finished_partitions.add(message.partition)
-                self.process_consumer_offset_message(message)
-            except ConsumerTimeout:
-                break
-            except (
-                    FailedPayloadsError,
-                    KafkaUnavailableError,
-                    LeaderNotAvailableError,
-                    NotLeaderForPartitionError,
-            ) as e:
-                self.retry += 1
-                self.log.warning("Got %s, retrying", e.__class__.__name__)
+            message = self.consumer.next()
```

Comment:
Message is None in case of timeout. Perhaps we can just validate that condition and throw an error.